File: seed.py
import sqlite3
import os

appdata = os.environ.get('APPDATA')
db_path = os.path.join(appdata, 'electron-react-boilerplate', 'speakspace.db')

conn = sqlite3.connect(db_path)
cursor = conn.cursor()

# Create workspace if empty
cursor.execute("SELECT id FROM workspaces LIMIT 1")
ws = cursor.fetchone()
if ws:
    ws_id = ws[0]
else:
    cursor.execute("INSERT INTO workspaces (name) VALUES ('Default Workspace')")
    ws_id = cursor.lastrowid
    print('Created workspace:', ws_id)

cursor.execute("""
INSERT INTO notes (workspace_id, name, audio_relative_path, transcript, is_pinned, created_at, updated_at)
VALUES (?, ?, ?, ?, ?, datetime('now', 'localtime'), datetime('now', 'localtime'))
""", (ws_id, '產品規劃會議', 'audio/meeting1.mp3', '今天會議我們決定將 SpeakSpace 的最新功能推進到下一個階段。實時轉錄和儀表板的串接已經完成，接下來準備讓團隊開始測試，確保本機端的 AI 推論穩定度。', 1))

print('Inserted note:', cursor.lastrowid)

conn.commit()
conn.close()

File: scripts/dev/seed.js
const Database = require('better-sqlite3');
const path = require('path');

// 产品改名后 userData 目录跟着 productName 走。
const dbPath = path.join(process.env.APPDATA, 'LetsVoice', 'letsvoice.db');
const db = new Database(dbPath);

console.log('Opened DB:', dbPath);

// Ensure a workspace exists
const ws = db.prepare('SELECT id FROM workspaces LIMIT 1').get();
let workspaceId;
if (ws) {
  workspaceId = ws.id;
} else {
  const insertWs = db.prepare('INSERT INTO workspaces (name) VALUES (?)');
  const info = insertWs.run('Default Workspace');
  workspaceId = info.lastInsertRowid;
  console.log('Created Default Workspace:', workspaceId);
}

// Insert a note
const insertNote = db.prepare(`
    INSERT INTO notes (workspace_id, name, audio_relative_path, transcript, is_pinned, created_at, updated_at)
    VALUES (?, ?, ?, ?, ?, datetime('now'), datetime('now'))
`);

const info2 = insertNote.run(
  workspaceId,
  '專案進度週會紀錄',
  'mock-audio.mp3',
  '今天的會議主要討論 Dashboard 的實作進度。目前已經順利把前端跟後端的資料庫串接好了，接下來要測試實時轉錄。大家覺得有沒有什麼需要補充的？',
  1,
);

console.log('Created Mock Note:', info2.lastInsertRowid);
db.close();
